apm-agent-core/src/main/java/co/elastic/apm/bci/ElasticApmAgent.java

```diff
@@ -51,6 +58,7 @@
 
 public class ElasticApmAgent {
 
+    private static final ConcurrentMap<Class<? extends ElasticApmInstrumentation>, MatcherTimer> matcherTimers = new ConcurrentHashMap<>();
```

Comment:
This can also be useful to debug customer problems. The matcher stats are logged on DEBUG when the application shuts down.

Reply:
In this case, it looks like a potential leak- sooner or later one of our Instrumentation classes will hold a reference to a helper class loaded by some contextual class loader, which will prevent an entire context to be collected. It can even happen with the JDBC or Servlet instrumentation if someone decides to package these API jars with their app and using a class loader that looks locally before delegating lookup upwards (e.g. the default Tomcat webapp CL, at least in older versions).
If this code is here, I would assign a unique ID for each instrumentation and map to that instead of the instrumentation class, even if it's just for debug purposes.

Reply:
good catch! I'll just use the class name then